SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp

```diff
@@ -225,12 +228,15 @@ void USpatialReceiver::OnAuthorityChange(const Worker_AuthorityChangeOp& Op)
 
 void USpatialReceiver::HandlePlayerLifecycleAuthority(const Worker_AuthorityChangeOp& Op, APlayerController* PlayerController)
 {
+	USpatialNetConnection* Connection = Cast<USpatialNetConnection>(PlayerController->GetNetConnection());
+	bool bIsServer = NetDriver->IsServer();
+
 	// Server initializes heartbeat logic based on its authority over the position component,
 	// client does the same for heartbeat component
-	if ((NetDriver->IsServer() && Op.component_id == SpatialConstants::POSITION_COMPONENT_ID) ||
-		(!NetDriver->IsServer() && Op.component_id == SpatialConstants::HEARTBEAT_COMPONENT_ID))
+	if ((bIsServer && Op.component_id == SpatialConstants::POSITION_COMPONENT_ID) ||
+		(!bIsServer && Op.component_id == SpatialConstants::HEARTBEAT_COMPONENT_ID))
 	{
-		if (Op.authority == WORKER_AUTHORITY_AUTHORITATIVE)
+		if (Op.authority == WORKER_AUTHORITY_AUTHORITATIVE && Connection != nullptr)
 		{
 			if (USpatialNetConnection* Connection = Cast<USpatialNetConnection>(PlayerController->GetNetConnection()))
```

Comment:
These if statements can be reworked now you have the connection and bIsServer

Reply:
Removing all together since the if statement above does the exact same check